src/internal/controllers/access_controller.go

```diff
@@ -23,29 +17,24 @@ func (c *Controller) CIsAdmin(curUserId string) models.Response {
 	}
 
 	return utils.ErrorForbidden("User is not admin")
-	*/
-	return utils.Response("TODO: Is admin")
 }
 
 func (c *Controller) CIsSuperAdmin(curUserId string) models.Response {
-	/*
-	response = c.CGetAdminById(curUserId)
-	if utils.IsTypeError(response) {
+	response := c.CGetAdminByUserId(curUserId)
+	if utils.IsTypeError(response) && response.Data.(models.Err).StatusCode != http.StatusNotFound{
 		return response
 	}
 
-	if utils.IsType(response, models.AdminOutput{}) && response.Data.(models.AdminOutput).Access == "superadmin"{
+	if utils.IsType(response, models.AdminOutput{}) && response.Data.(*models.AdminOutput).Access == "superadmin"{
```

Comment:
När jag testade i Swagger så kunde den inte hitta att min user var superadmin utan att jag ändrade denna raden till att ha pointers i denna check. Det är funktionen CIsSuperAdmin i access_controller.go

```suggestion
	if utils.IsType(response, &models.AdminOutput{}) && response.Data.(*models.AdminOutput).Access == "superadmin"{
```

Reply:
Det fungerar för mig utan &, jag la till i utils.IsType att ge true om det är rätt modell eller pointer för rätt modell.